src/libcore/mem.rs

```diff
@@ -180,6 +180,31 @@ pub fn forget<T>(t: T) {
 /// More specifically, this is the offset in bytes between successive
 /// items of the same type, including alignment padding.
 ///
+/// In general, the size of a type is not stable between compiler releases. But for primitive types
+/// and Option, the size is stable.
+///
+/// The following table gives the size for types where the return value is stable. `T` is a sized
+/// type while `UT` is an unsized type.
+///
+/// Type | size_of<Type>() return value
+/// ---- | ----------------------------
+/// () | 0
+/// u8 | 1
+/// u16 | 2
+/// u32 | 4
+/// u64 | 8
+/// i8 | 1
+/// i16 | 2
+/// i32 | 4
+/// i64 | 8
+/// usize | pointer size
+/// isize | pointer size
+/// [T, n] | `size_of::<T>() * n`
+/// &T | pointer size
+/// &UT | 2 * pointer size
```

Comment:
I think the accepted `extern types` RFC means that there will exist types that are not `Sized` but are none-the-less thin pointers.  And there has been discussion of extra-fat pointers for things like `dyn (Debug + Add)`.  So it might be better to not mention this particularly, especially since use cases for relying on it are kinda sketchy.

Reply:
So basically the size of `&T` is generally unstable?